testing/e2e/suites/nginxIngressController.go

```diff
@@ -187,6 +204,101 @@ func nicTests(in infra.Provisioned) []test {
 					return err
 				}
 
+				lgr.Info("finished testing")
+				return nil
+			},
+		},
+		{
+			name: "testing DefaultSSLCertificate validity",
+			cfgs: builderFromInfra(in).
+				withOsm(in, false, true).
+				withVersions(manifests.OperatorVersionLatest).
+				withZones([]manifests.DnsZoneCount{manifests.DnsZoneCountNone}, []manifests.DnsZoneCount{manifests.DnsZoneCountNone}).
+				build(),
+			run: func(ctx context.Context, config *rest.Config, operator manifests.OperatorConfig) error {
+				lgr := logger.FromContext(ctx)
+				lgr.Info("starting test")
+
+				c, err := client.New(config, client.Options{
+					Scheme: scheme,
+				})
+				if err != nil {
+					return fmt.Errorf("creating client: %w")
+				}
+
+				// get keyvault uri
+				kvuri := in.Cert.GetId()
+
+				// create defaultSSLCert
+				defaultSSLCert := v1alpha1.DefaultSSLCertificate{
+					KeyVaultURI: &kvuri,
+				}
+
+				// create nic and upsert
+				testNIC := manifests.NewNginxIngressController("nginx-ingress-controller", "nginxingressclass")
+				testNIC.Spec.DefaultSSLCertificate = &defaultSSLCert
+				if err := upsert(ctx, c, testNIC); err != nil {
+					return fmt.Errorf("upserting NIC: %w", err)
+				}
+
+				var service v1alpha1.ManagedObjectReference
```

Comment:
why are we waiting for the service?

Reply:
instead wait for the Nic to have a status of available. That should mean the spc is available too so checking the spc wouldn't have to go into a retry if you wait for the nic to be available